apps/www/_blog/2023-12-15-v2-client-libraries.mdx

```diff
@@ -0,0 +1,170 @@
+
+<div className="video-container">
+  <iframe
+    className="w-full"
+    src="https://www.youtube-nocookie.com/embed/7DYZdAPPJOw"
+    title="YouTube video player"
+    allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
+    allowfullscreen
+  />
+</div>
+
+By default, `supabase-js` uses the browser's `localStorage` mechanism to persist the user's session. This can be extended with platform-specific implementations. React Native can target native mobile and web applications with the same code base, so we need a storage implementation that works for all these platforms. Now you can use [react-native-async-storage](https://supabase.com/blog/react-native-authentication#authentication-storage) or a combination with [expo-secure-sorage](https://supabase.com/blog/react-native-authentication#encrypting-the-user-session) for AES encrypted sessions.
+
+Beyond that we’ve focused on making supabase-js highly compatible with React Native and created plenty of [video tutorials](https://youtube.com/playlist?list=PL5S4mPUpp4OsrbRTx21k34aACOgpqQGlx&si=Ez-0S4QhBxtayYsq) and documentation for [crafting an offline-first experience](https://supabase.com/blog/react-native-offline-first-watermelon-db), [uploading files to Supabase Storage](https://supabase.com/blog/react-native-storage), and [sending push notifications](https://supabase.com/docs/guides/functions/examples/push-notifications).
+
+## Our approach to client libraries
+
+We have a strong preference to develop the client libraries with our community. This is part of our open source philosophy:
+
+### The Cathedral or the Bazaar?
+
+If you haven’t already, it’s worth reading “[The Cathedral and the Bazaar](https://en.wikipedia.org/wiki/The_Cathedral_and_the_Bazaar)” by Eric S. Raymond. In short, it contrasts two software development approaches:
+
+1. **The "Cathedral"** represents closed, centralized development, where a small group of developers work in isolation.
+2. **The "Bazaar"** symbolizes open, decentralized collaboration, where a large community of developers openly shares and collaborates on code.
+
+We believe the “bazaar” model is the right model for an open source business. If you aren’t constantly pushing in this direction then it’s extremely likely that the company will relicense (seen most recently with [Hashi](https://news.ycombinator.com/item?id=37081306)). The way we see it, the more we can foster our community the less power we have.
+
+### Fostering the community
+
+We just reached 1000 contributors to our [main repo](https://github.com/supabase/supabase). It’s not easy to build a community of contributors, it’s something that need to be fostered.
+
+The client libs are one of the best ways to foster the community because they lower complexity than some of the tools we maintain (do you know [Haskell](https://github.com/PostgREST/postgrest)?).
```

Comment:
```suggestion
The client libs are one of the best ways to foster the community because they are lower-complexity than some of the tools we maintain (do you know [Haskell](https://github.com/PostgREST/postgrest)?).
```